backend/src/database/repositories/segmentRepository.ts

```diff
@@ -182,39 +182,62 @@ class SegmentRepository extends RepositoryBase<
         'sourceId',
         'sourceParentId',
         'customActivityTypes',
-        'activityChannels',
       ].includes(key),
     )
 
-    let segmentUpdateQuery = `UPDATE segments SET `
-    const replacements = {} as any
+    if (updateFields.length > 0) {
+      let segmentUpdateQuery = `UPDATE segments SET `
+      const replacements = {} as any
 
-    for (const field of updateFields) {
-      segmentUpdateQuery += ` "${field}" = :${field} `
-      replacements[field] = data[field]
+      for (const field of updateFields) {
+        segmentUpdateQuery += ` "${field}" = :${field} `
+        replacements[field] = data[field]
 
-      if (updateFields[updateFields.length - 1] !== field) {
-        segmentUpdateQuery += ', '
+        if (updateFields[updateFields.length - 1] !== field) {
+          segmentUpdateQuery += ', '
+        }
       }
-    }
 
-    segmentUpdateQuery += ` WHERE id = :id and "tenantId" = :tenantId `
-    replacements.tenantId = this.options.currentTenant.id
-    replacements.id = id
+      segmentUpdateQuery += ` WHERE id = :id and "tenantId" = :tenantId `
+      replacements.tenantId = this.options.currentTenant.id
+      replacements.id = id
 
-    if (replacements.customActivityTypes) {
-      replacements.customActivityTypes = JSON.stringify(replacements.customActivityTypes)
-    }
+      if (replacements.customActivityTypes) {
+        replacements.customActivityTypes = JSON.stringify(replacements.customActivityTypes)
+      }
 
-    if (replacements.activityChannels) {
-      replacements.activityChannels = JSON.stringify(replacements.activityChannels)
+      await this.options.database.sequelize.query(segmentUpdateQuery, {
+        replacements,
+        type: QueryTypes.UPDATE,
+        transaction,
+      })
     }
 
-    await this.options.database.sequelize.query(segmentUpdateQuery, {
-      replacements,
-      type: QueryTypes.UPDATE,
-      transaction,
-    })
+    if (data.activityChannels && typeof data.activityChannels === 'object') {
+      if (Object.keys(data.activityChannels).length > 0) {
+        let activityChannelsUpsertQuery = `INSERT INTO "segmentActivityChannels" ("tenantId", "segmentId", "platform", "channel") VALUES `
+        for (const platform in data.activityChannels) {
+          if (Object.prototype.hasOwnProperty.call(data.activityChannels, platform)) {
+            for (let i = 0; i < data.activityChannels[platform].length; i++) {
+              const channel = data.activityChannels[platform][i]
+              activityChannelsUpsertQuery += `(:tenantId, :segmentId, '${platform}', '${channel}'),`
```

Comment:
That's dangerous, let's do something like [here](https://github.com/CrowdDotDev/crowd.dev/blob/165662c5f28688f5232ac8d5c4615abd2019d7ba/backend/src/database/repositories/memberSegmentAffiliationRepository.ts#L86C1-L112) instead